pkg/jobcreator/controller.go

```diff
@@ -319,9 +322,23 @@ func (controller *JobCreatorController) checkResults() error {
 			// there is an error with the job
 			// accept anyway
 			// TODO: trigger mediation here
-			controller.acceptResult(dealContainer)
+			err := controller.acceptResult(dealContainer)
+			if err != nil {
+				controller.log.Error("failed to accept results", err)
+				return err
+			}
 		} else {
-			controller.downloadResult(dealContainer)
+			// We check for all completed deals, including deals whose results
+			// we have already downloaded. Check the download path and download
+			// if results do not exist.
+			downloadPath := solver.GetDownloadsFilePath(dealContainer.ID)
+			if _, err := os.Stat(downloadPath); errors.Is(err, fs.ErrNotExist) {
```

Comment:
Common advice online recommends using `os.IsNotExist`, but the docs for this function recommend that new code should use `errors.Is(err, fs.ErrNotExist)`: https://pkg.go.dev/os#IsNotExist